### PR TITLE
Fix PowerVS cluster's load balancer status check

### LINKED ISSUE #2026: Cluster incorrectly shown as ready when both LoadBalancers are not ready
/kind bug
/area provider/ibmcloud

**What steps did you take and what happened:**
Deploy a 4.18 cluster on a PowerVS zone where LoadBalancers are slow to create.
We are called with `InfraReady`.  We then create DNS records for the LBs.  However, only the public LB exists.  So the cluster fails to deploy.

**What did you expect to happen:**
We should wait for all specified LBs to become ready.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Cluster-api version:
- Minikube/KIND version:
- Kubernetes version: (use `kubectl version`):
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by dharaneeshvrd**
Need to depend on load balancer's state to set the cluster's ready status instead of using requeue logic which is designed to block the reconciliation.

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #2026 

**Special notes for your reviewer**:

/area provider/ibmcloud

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

### CODE REVIEW

File: cloud/scope/powervs_cluster.go [hunk 1]
```diff
@@ -1974,8 +1974,9 @@ func (s *PowerVSClusterScope) ReconcileLoadBalancers() (bool, error) {
 				return false, err
 			}
 
-			if requeue := s.checkLoadBalancerStatus(*loadBalancer); requeue {
-				return requeue, nil
+			if isReady := s.checkLoadBalancerStatus(*loadBalancer); !isReady {
+				s.V(3).Info("LoadBalancer is still not Active", "name", *loadBalancer.Name, "state", *loadBalancer.ProvisioningStatus)
```
Comment (Karthik-K-N) [inline cloud/scope/powervs_cluster.go:1978]: *loadbalancer.Name can be nil when user sets only id. so better avoid it.

File: cloud/scope/powervs_cluster.go [hunk 2]
```diff
@@ -1987,26 +1988,14 @@ func (s *PowerVSClusterScope) ReconcileLoadBalancers() (bool, error) {
 			continue
 		}
 
-		// check VPC load balancer exist in cloud
-		loadBalancerStatus, err := s.checkLoadBalancer(loadBalancer)
```
Comment (Karthik-K-N) [inline cloud/scope/powervs_cluster.go:1991]: why is this removed?

File: cloud/scope/powervs_cluster.go (hunk 1 above)
Comment (dharaneeshvrd) [reply cloud/scope/powervs_cluster.go:1978]: It's coming from cloud API resp right, would be able to create a LB without name?

Comment (Karthik-K-N) [reply cloud/scope/powervs_cluster.go:1978]: oh yeah, sorry, I overlooked loadbalancer from
```
for index, loadBalancer := range loadBalancers {
```

File: cloud/scope/powervs_cluster.go (hunk 2 above)
Comment (dharaneeshvrd) [reply cloud/scope/powervs_cluster.go:1991]: `checkLoadBalancer` checks load balancer by the user provided name, which we already do in prev block [here](https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/blob/6fc82480e3361b2658699075690765dc66966c1f/cloud/scope/powervs_cluster.go#L1963) by recently added code via [this](https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/commit/3edcccbf57561ee7b6498b7f56fcf304bb9f0c19).

Comment (Karthik-K-N) [reply cloud/scope/powervs_cluster.go:1991]: `s.GetLoadBalancerID(loadBalancer.Name)` just checks only from status not from cloud!

Comment (dharaneeshvrd) [reply cloud/scope/powervs_cluster.go]: Sry, did not look properly, reverted the changes.

**Review by Karthik-K-N — COMMENTED** [review]
/lgtm
/assign @Amulyam24 @mkumatag

**Review by Amulyam24 — COMMENTED** [review]
/lgtm

Thank you for the quick fix @dharaneeshvrd!

**Comment by Karthik-K-N** [comment]
/hold

@KeerthanaAP is doing some testing, will update here once done.

**Comment by Karthik-K-N** [comment]
Thanks @KeerthanaAP for testing it quickly.
@dharaneeshvrd i think we discussed about this but forgot , we need to make sure all the LBs are active before infra is set to true.

**Comment by Karthik-K-N** [comment]
> @KeerthanaAP Can you please double check this? It is working for me when I tested in local and in the log you have provided for this deployment does not have my changes.

Checking with @ hamzy  about this in slack

> @Karthik-K-N I have considered and did the changes earlier and it was working properly, but it was creating the LBs serially, now I have added a flag to create them without blocking further LBs reconcile. PTAL now.
Got it earlier will work but there was no gurantee that in case of multiple LBs all will be in Active as we used to return once we found one LB in active. But I think with this change we should be good

**Review by Karthik-K-N — COMMENTED** [review]
xIts been [verified](https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/issues/2026#issuecomment-2442438100) and works as expected.
/lgtm

**Review by mkumatag — APPROVED** [review]
/unhold

**Pull request merged**